verifyml/model_tests/FEAT/tests/test_MinMaxMetricThreshold.py

```diff
@@ -0,0 +1,148 @@
+        "prediction": [1, 1, 1, 0.8],
+    }
+)
+
+
+def test_plot_defaults():
+    """Test that the default arguments of the plot() method are as expected."""
+
+    sig = inspect.signature(MinMaxMetricThreshold.plot)
+
+    assert sig.parameters["alpha"].default == 0.05
+    assert sig.parameters["save_plots"].default == True
+
+
+def test_save_plots_true_classification():
+    """Test that the plot is saved to the test object when .plot(save_plots=True)."""
+    # init test object
+    test_obj = MinMaxMetricThreshold(attr="gender", metric="fpr", threshold=0.025,)
+
+    # run test
+    test_obj.run(test_data_classification)
+
+    # plot it
+    test_obj.plot(save_plots=True)
+
+    # test object should be a dict of length 1
+    assert len(test_obj.plots) == 1
+
+    # test object should have the specified key, and the value should be a string
+    assert isinstance(test_obj.plots["ROC Curve of gender groups"], str)
+
+
+def test_save_plots_true_regression():
+    """Test that the plot is saved to the test object when .plot(save_plots=True)."""
+    # init test object
+    test_obj = MinMaxMetricThreshold(attr="gender", metric="mse", threshold=0.5,)
+
+    # run test
+    test_obj.run(test_data_regression)
+
+    # plot it
+    test_obj.plot(save_plots=True)
+
+    # test object should be a dict of length 1
+    assert len(test_obj.plots) == 1
+
+    # test object should have the specified key, and the value should be a string
+    assert isinstance(test_obj.plots["Mean Squared Error across gender subgroups"], str)
+
+
+def test_save_plots_false_classification():
+    """Test that the plot is not saved to the test object when .plot(save_plots=False)."""
+    # init test object
+    test_obj = MinMaxMetricThreshold(attr="gender", metric="fnr", threshold=0.5,)
+
+    # run test
+    test_obj.run(test_data_classification)
+
+    # plot it
+    test_obj.plot(save_plots=False)
+
+    # nothing should be saved
+    assert len(test_obj.plots) == 0
+
+
+def test_save_plots_false_regression():
+    """Test that the plot is not saved to the test object when .plot(save_plots=False)."""
+    # init test object
+    test_obj = MinMaxMetricThreshold(attr="gender", metric="mae", threshold=0.5,)
+
+    # run test
+    test_obj.run(test_data_regression)
+
+    # plot it
+    test_obj.plot(save_plots=False)
+
+    # nothing should be saved
+    assert len(test_obj.plots) == 0
+
+
+def test_run_classification():
+    """Test that calling .run() updates the test object's .result and .passed attributes."""
+    # init test object
+    test_obj1 = MinMaxMetricThreshold(attr="gender", metric="fpr", threshold=0.5,)
+
+    test_obj2 = MinMaxMetricThreshold(attr="gender", metric="fnr", threshold=0.7,)
+
+    # run tests
+    test_obj1.run(test_data_classification)
+    test_obj2.run(test_data_classification)
+
+    assert (
+        test_obj1.result.loc["gender_F"]["fpr at current probability threshold"]
+        == 0.333
+    )  # rounded to 3 d.p
```

Comment:
Do we round to 3 d.p. by default? Alternatively for approximate matching there is a `pytest.approx()` function.

Reply:
yup, the feat test will rounds all final outputs to 3 d.p.